packages/paste-core/components/form-pill-group/src/FormPill.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {CompositeItem} from '@twilio-paste/reakit-library';
+import type {CompositeStateReturn} from '@twilio-paste/reakit-library';
+import {PillCloseIcon} from './PillCloseIcon';
+
+interface FormPillStylesProps {
+  selected?: boolean;
+  element?: string;
+  children: React.ReactNode;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+  onClick?: () => void;
```

Comment:
Could we instead change this to something like one of the following?

`onClick?: HTMLButtonElement['onClick'];`
`onClick?: `MouseEventHandler<HTMLButtonElement>;`
`onClick?: (event: React.ClickEvent<HTMLButtonElement>) => void;`

I ask because in case our consumers want to use the event in their `onClick` callback they pass to this component. As it is typed, I think this would discourage that.

Reply:
This component (and prop) isn't actually exposed to the user. This is used by the `FormPill` in the same file, which passes in the `onSelect` as an `onClick`